protocol/lib/error/logging.go

```diff
@@ -40,14 +43,16 @@ func WrapErrorWithSourceModuleContext(err error, module string) error {
 		WithLogKeyValue(SourceModuleKey, fmt.Sprintf("x/%v", module))
 }
 
-// LogErrorWithBlockHeight logs an error, appending the block height and ABCI callback to the error message.
-func LogErrorWithBlockHeight(logger log.Logger, err error, blockHeight int64, callback string) {
+// LogDeliverTxError logs an error, appending the block height and ABCI callback to the error message.
+func LogDeliverTxError(logger log.Logger, err error, blockHeight int64, handler string, msg sdk.Msg) {
 	if err != nil {
 		err = errorsmod.Wrapf(
 			err,
-			"Block height: %d, Callback: %s",
+			"Block height: %d, Handler: %s, Callback: %s, Msg: %+v",
 			blockHeight,
-			callback,
+			handler,
+			metrics.DeliverTx,
+			msg,
 		)
 		logger.Error(err.Error())
```

Comment:
nit: let's not format this  - instead we should do 

logger.Error(
    "error string", 
    "key", value,
    "key1", value1,
    ....
)

this makes it easier to filter/search on datadog